packages/component-library/components/types.ts

```diff
@@ -0,0 +1,64 @@
+import { AriaAttributes } from "react"
+
+export type Safelist = AriaAttributes & {
+  id?: string
+  itemProp?: string
+  itemScope?: string
+  itemType?: string
+  itemID?: string
+  itemRef?: string
+  tabIndex?: string
+  title?: string
+  role?: string
+}
+
+export type GridFractions =
+  | 0
+  | 0.25
+  | 0.5
+  | 0.75
+  | 1
+  | 1.25
+  | 1.5
+  | 1.75
+  | 2
+  | 2.5
+  | 3
+  | 3.5
+  | 4
+
+export type Breakpoint = {
+  mobile: "0"
+  tablet: "768"
+  desktop: "1024"
+}
```

Comment:
It feels like we're in the same boat here as what we talked about in retro with the last PR. I want to approve this PR, but I don't feel confident yet that we came to a consensus on the breakpoints. This could just be me missing/forgetting the memo though, or not making sure I'm across the research you've done already. @ActuallyACat Do you feel like you have a solid idea of what people want and what the next steps are if we're going to modify these? Can we actually sign off on this as it is and stand behind it with good reasons?

Reply:
The page I was on:
- no need to provide a way to do custom breakpoints. We can cross this bridge when people need it, and we have several remaining to explore (such as dynamic stylesheets, as P.O.C'd in the last PR)
- But, I was expecting that we would ditch the device specific breakpoints in favor or abstract ones, seeing as there was general consensus in the practice that this was better from the poll. We could change it later but that could be difficult if people are already relying on the device specific ones.

Reply:
Both good points.

As for the naming, why have names at all? If we're worried about it being difficult to change later, why not just refer to them by their pixel values instead of choosing a name? I mean, if TypeScript is going to show you what options you have to choose from, it sort of eliminates the problem of having to remember what they are, right?

Reply:
This is my opinion:

- Heading component: **essential**
- Spacing component: **essential** (we will definitely need spacing between text, and we don't want people hacking it with custom div wrappers and custom classes)
- responsive spacing: **not essential**  (it's cool, but I never saw this as a requirement, and I never saw a design for it - should spacing necessarily get tighter on smaller screens? Perhaps we could just remove it in this PR, and do a future follow up PR?)

I guess what I'm saying

Reply:
I think the issue is that if people use this spacing component as their spacing solution, they have no way to make it responsive because they can't use CSS anymore.

In other words, we still need to build that functionality before people can start using the spacing component (to avoid nasty surprises where they can't make it responsive), but if we just want to ship this to start working with it in the component library, I'd have no problem with removing it from this PR and postponing it until a future one.

Reply:
Are there many examples of responsive spacing at present? If it's currently used often, then we should support it. If not, is this an imagined requirement?

Reply:
> It feels like we're in the same boat here as what we talked about in retro with the last PR. I want to approve this PR, but I don't feel confident yet that we came to a consensus on the breakpoints. This could just be me missing/forgetting the memo though, or not making sure I'm across the research you've done already. @ActuallyACat Do you feel like you have a solid idea of what people want and what the next steps are if we're going to modify these? Can we actually sign off on this as it is and stand behind it with good reasons?

Going to rip this out of the PR and open up a Box PR separately... This PR was opened before I decided to remove spacing out of the other PR.